Core/IScopedCache.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace PubComp.Caching.Core
+{
+    public interface IScopedCache : ICache
```

Comment:
Why not simply use ScopedValue<TValue> as the value with ICache, what's the added value of adding ScopedValue and IScopedCache to the infrastructure?

Reply:
I've created 3 new cache types:
- InMemory**Scoped**Cache
- Redis**Scoped**Cache
- Layered**Scoped**Cache
which by using the new ScopedValue and CacheDirectives
gives the consumer the possibility to specify the maximum age (using MinimumValueTimestamp)
based on its own specific BL

Use cases for example:
**Service A** provides CustomerDetails

**Service B** needs CustomerDetails.RegistrationDate which doesn't change over time so cache result is always accepted
**Service C** has a BL which can accept CustomerDetails retrieved in the last 30 minutes
**Service D** has a BL which can accept CustomerDetails retrieved at least after a specific timestamp
**Service E** BL requires up-to-date CustomerDetails (no-cache)

by using CacheableActionFilterAttribute and *ScopedCache defined with slidingExpiration of 30 minutes we can serve all listed above **easily** and **safely** using a single endpoint in **Service A**

* as a bonus, it makes it easier to retrieve data from replicated datasources while taking into consideration replication lags by relying on ValueTimestamp and MinimumValueTimestamp and GetScoped/SetScoped added functionalities

Reply:
Wouldn't each of these 4 services be using their own distinct cache name to enforce different full key names and therefore also use their own cache policy to make sure they don't get stale data according to their own definition of stale?

Reply:
no, the idea of scoped cache is to help service A


in the scenario above
services b-d are going to consume the data once
while supplying service A directives as HINT


if one of them needs it more then once, they will use their own cache